### DISCUSSION TITLE
How to test Infolist Action?

### DISCUSSION THREAD

**Discussion by brunomdsilva** [filamentphp/filament#12127] [Help] [answered]
### Package

Infolist builder

### Package Version

v3.2

### How can we help you?

I have a infolist with a ActionsGroup, I want to test if those buttons are working, but I can't figure out how to trigger those actions in the tests. (they work normally in the app)

Infolist Action:
```php
Action::make('cancelBooking')
  ->color('danger')
  ->requiresConfirmation()
  ->disabled(function (Booking $record) {
    return $record->cancelled_at || $record->entered_at || $record->exited_at;
  })
  ->action(function (Booking $record) {
    $record->cancelled_at = now();
    $record->save();

    Notification::make()
      ->title('Booking cancelled')
      ->success()
      ->send();
    }),
```

Test:
```php
livewire(ViewBooking::class, ['record' => $booking->getRouteKey()])
  ->callInfolistAction('action', 'cancelBooking');

  expect($booking->status)->toBe(BookingStatus::CANCELLED);
```

With that test I get:
Call to a member function getKey() on null

The button does exists in the test because I "assertSee" it and it passes, but I cannot call it's action.

I've tried `mountInfolistAction` before `callMountedInfolistAction`, calling like: `call`, `callAction`, `callInfolistAction` etc... I really can't figure out how to do this

**Comment by leandrocfe** [comment]
```php
livewire(ViewBooking::class, [
    'record' => $booking->getRouteKey(),
])
    ->mountAction('cancelBooking')
    ->callMountedAction();
```
?

**Reply by brunomdsilva** [reply]
This way there is no error, but it also doesn't work, `mountInfolistAction` also gave this result:
![image](https://github.com/filamentphp/filament/assets/56403619/8f51d16a-6048-4ad0-ba24-9c13db3a6e96)

**Reply by leandrocfe** [reply]
Are you sure that the update function in the action is working?

**Reply by brunomdsilva** [reply]
Yes, just tested again, it is notifying and being updated in the database as expected

**Reply by leandrocfe** [reply]
Could you try this removing the `->disabled()` function?

**Reply by leandrocfe** [reply]
You can also add `->assertActionEnabled('cancelBooking')` to check if the action is enabled..

**Reply by leandrocfe** [reply, marked as answer]
```php
->call('mountInfolistAction', 'cancelBooking', '.cancelBookingAction', 'infolist')
```

**Comment by brunomdsilva** [comment]
using helpers:

```php
->mountInfolistAction('. cancelBookingAction', 'cancelBooking')
->callMountedInfolistAction();
```
